modules/git/repo.go

```diff
@@ -125,7 +125,7 @@ func OpenRepository(repoPath string) (*Repository, error) {
 
 // Close this repository, in particular close the underlying gogitStorage if this is not nil
 func (repo *Repository) Close() {
-	if repo.gogitStorage == nil {
+	if repo == nil || repo.gogitStorage == nil {
```

Comment:
This is weird. Can an object method in golang be called even when the object is nil? (e.g. in C++ you can't). I mean, is it possible for repo to be `nil` here?

Reply:
It appears possible - The reason why I had to add it is because I had put `==` when I meant `!=` in an earlier iteration however, it genuinely appears to be the case:

https://play.golang.org/p/nCrm8x910Wt

I kinda feel it's ok to leave it in there.